### PR TITLE
Feature post

### PR DESCRIPTION
**Pull request by OmKakatkar**
# Feature: Post CRUD
## Additions
* Posts can be read, created, edited and deleted.
* Adds 2 dummy posts

### CODE REVIEW

File: src/frontend/components/PostCard/PostCard.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useSelector } from "react-redux";
+import useDetectClickOutside from "../../hooks/useDetectClickOutside";
+import ModalCard from "../ModalCard/ModalCard";
+import PostCardMenu from "../PostCardMenu/PostCardMenu";
+import "./PostCard.css";
+
+function PostCard({ post }) {
+	const { user } = useSelector((store) => store.auth);
+	const { triggerRef, nodeRef, showItem } = useDetectClickOutside();
```
Comment (MarkVed17) [inline src/frontend/components/PostCard/PostCard.jsx:19]: Learnt an alternate way to useOnClickOutside.

File: src/frontend/constants/api-constant.js [hunk 2]
```diff
@@ -4,3 +4,6 @@ export const API_USER_EDIT = "/api/users/edit";
 export const API_USERS = "/api/users/";
 export const API_FOLLOW = "/api/users/follow";
 export const API_UNFOLLOW = "/api/users/unfollow";
+export const API_POST = "/api/posts";
```
Comment (MarkVed17) [inline src/frontend/constants/api-constant.js:7]: Constants for API endpoints. Never thought of this.

**Review by MarkVed17 — APPROVED** [review]
I liked the coding conventions practised.
Looks good to proceed!

File: src/frontend/features/Post/Post.css [hunk 3]
```diff
@@ -0,0 +1,13 @@
+.page-overlay {
+	position: fixed;
+	background-color: var(--clr-overlay);
+	z-index: 10;
+	top: 0;
```
Comment (mansipadgelwar) [inline src/frontend/features/Post/Post.css]: can use inset instead

**Review by mansipadgelwar — APPROVED** [review]
Good use of constants, naming conventions, variables, and comments.

File: src/frontend/features/Post/Post.css (hunk 3 above)
Comment (OmKakatkar) [reply src/frontend/features/Post/Post.css]: Done

**Pull request merged**